polkadot/runtime/parachains/src/inclusion/tests.rs

```diff
@@ -1582,8 +1574,10 @@ fn candidate_checks() {
 			let candidate_receipt_with_backers = expected
 				.entry(backed_candidate.hash())
 				.or_insert_with(|| (backed_candidate.receipt(), Vec::new()));
-			let (validator_indices, _maybe_core_index) =
-				backed_candidate.validator_indices_and_core_index(true);
+			let (validator_indices, Some(_)) = backed_candidate.validator_indices_and_core_index()
+			else {
+				panic!("Expected core index")
```

Comment:
validator_indices_and_core_index can return None for core index. Likely not possible in practice, but are we still testing it anywhere ?

Reply:
I'll add a test for it. Such candidates should be dropped if they don't contain a core index commitment

Reply:
Done, see `candidate_without_core_index` in paras_inherent tests